src/Functions/TransitFunctionApp/QueryDevices.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Microsoft.Azure.Devices;
+using System.Text;
+
+namespace Transportation.Demo.Functions
+{
+    public static class QueryDevices
+    {
+        [FunctionName("QueryDevices")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            var registryManager = RegistryManager.CreateFromConnectionString(Environment.GetEnvironmentVariable("IotHubConnectionString"));
+
+            StringBuilder deviceQuery = new StringBuilder("select * from devices where status='enabled'");
+
+            if (req.Query.ContainsKey("id")) {
```

Comment:
we should validate input parameters for basic type validation and possibly to help avoid any injection problems (the most common form of attach).

Reply:
I agree. But what are the restrictions for the id, type, and location?

Reply:
Done! except for 'device types'. Where can I find the list of allowed device types?